librosa/core/constantq.py

```diff
@@ -1375,38 +1378,36 @@ def griffinlim_cqt(
         fmin = note_to_hz("C1")
 
     if random_state is None:
-        rng = np.random
+        rng = np.random.default_rng()
     elif isinstance(random_state, int):
-        rng = np.random.RandomState(seed=random_state)
+        rng = np.random.RandomState(seed=random_state)  # type: ignore
     elif isinstance(random_state, np.random.RandomState):
-        rng = random_state
+        rng = random_state  # type: ignore
```

Comment:
mypy isn't great at handling the randomstate object / random generator objects

Reply:
The type for `random_state` includes `np.random.Generator` as well, which isn't covered here, unless it is covered indirectly via inheritance. At least pyright complains that `rng` is possibly unbound though.

This is one of the cases where the type checkers exhaustiveness check via [`assert_never`](https://typing.readthedocs.io/en/latest/source/unreachable.html?highlight=assert_never#assert-never-and-exhaustiveness-checking) (from `typing_extensions` for backwards compatibility) could come in handy. In this case just an

```py
    else:
        assert_never(random_state)
```

This would guarantee that when extending the code, one cannot forget to think about these `if`/`elif` branches.

Another minor tweak that would help mypy is to actually type `rng` as `Callable[[int], np.ndarray]`, i.e., pull out the actual call to `.random` into a lambda that is easier to "unify" from a type system perspective.

All not crucial of course...

Reply:
Good catch - even better is to raise an exception for a bad random state type.  (Adding now.)

Reply:
> Good catch - even better is to raise an exception for a bad random state type. (Adding now.)

That is what `assert_never` does at runtime anyway, plus it ensures that the code is unreachable at type check time already. I'd really recommend to use `assert_never` in such cases, it has helped our team tremendously in factorings and is one of the features we enjoy the most about typed Python. It not only works for exhaustiveness checks of `Union` types, but also `Literal`s and `Enum` (you have to use `is` comparison with enums though instead of `==` to make it work though).

Reply:
I still think an explicit exception in an else clause is preferable here.  Assertion failures are no help to users, both in the sense of meaningful error messages, and in the sense of isolating failures at runtime by exception types.  It seems like the static analyzers are smart enough to detect `else: raise...` anyway.

Reply:
You could still give the user a `ParameterExcetion` with exactly that error message if you want to. I can demonstrate in my other PR...

The point is that the bug of forgetting to include in `np.random.Generator` in one of the if branches can easily be found by a type checker and shouldn't require any mental efforts to find.